Mathlib/GroupTheory/Submonoid/MulOpposite.lean

```diff
@@ -0,0 +1,49 @@
+/-
+Copyright (c) 2023 Eric Wieser. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Wieser, Jujian Zhang
+-/
+import Mathlib.GroupTheory.Submonoid.Basic
+import Mathlib.Algebra.Group.Opposite
+
+/-!
+# Submonoid of opposite monoids
+
+For every monoid `M`, we construct an equivalence between submonoids of `M` and that of `Mᵐᵒᵖ`.
+
+-/
+
+variable {M : Type*} [MulOneClass M]
+
+namespace Submonoid
+
+/-- pull a submonoid back to an opposite submonoid along `unop`-/
+@[to_additive (attr := simps) "pull an additive submonoid back to an opposite submonoid along
+`unop`"]
+def op {M : Type*} [MulOneClass M] (x : Submonoid M) : Submonoid (MulOpposite M) where
+  carrier := MulOpposite.unop ⁻¹' x.1
+  mul_mem' ha hb := x.mul_mem hb ha
+  one_mem' := Submonoid.one_mem' _
+
+/-- pull an opposite submonoid back to a submonoid along `op`-/
+@[to_additive (attr := simps) "pull an opposite additive submonoid back to a submonoid along `op`"]
+def unop {M : Type*} [MulOneClass M] (x : Submonoid (MulOpposite M)) : Submonoid M where
+  carrier := MulOpposite.op ⁻¹' x.1
+  mul_mem' ha hb := x.mul_mem hb ha
+  one_mem' := Submonoid.one_mem' _
```

Comment:
I guess it probably makes things less painful downstream if these are `protected`